com.microsoft.mrtk.input/Controllers/ArticulatedHandController.cs

```diff
@@ -147,33 +140,61 @@ protected override void UpdateTrackingInput(XRControllerState controllerState)
         {
             base.UpdateTrackingInput(controllerState);
 
-            // Workaround for missing aim/pointing-pose on devices without interaction profiles
-            // for hands, such as Varjo and Quest. Should be removed once we have universal
-            // hand interaction profile(s) across vendors.
+            // In case the position input action is not provided, we will try to polyfill it with the device position.
+            // Should be removed once we have universal hand interaction profile(s) across vendors.
+
             if (handsAggregator != null && (positionAction.action?.controls.Count ?? 0) == 0)
             {
-                if (!handsAggregator.TryGetJoint(TrackedHandJoint.IndexProximal, HandNode, out HandJointPose knuckle))
-                {
-                    return;
-                }
-                if (!handsAggregator.TryGetJoint(TrackedHandJoint.Palm, HandNode, out HandJointPose palm))
+                if (TryGetPolyfillDevicePose(out Pose devicePose))
                 {
-                    return;
-                }
+                    controllerState.position = devicePose.position;
+                    controllerState.rotation = devicePose.rotation;
+
+                    // Polyfill the tracking state, too.
+                    controllerState.inputTrackingState = InputTrackingState.Position | InputTrackingState.Rotation;
+                }                
+            }
+        }
+
+
+        private static readonly Quaternion rightPalmOffset = new Quaternion(Mathf.Sqrt(0.125f), Mathf.Sqrt(0.125f), -Mathf.Sqrt(1.5f) / 2.0f, Mathf.Sqrt(1.5f) / 2.0f);
+        private static readonly Quaternion leftPalmOffset = new Quaternion(Mathf.Sqrt(0.125f), -Mathf.Sqrt(0.125f), Mathf.Sqrt(1.5f) / 2.0f, Mathf.Sqrt(1.5f) / 2.0f);
+
+        // Workaround for missing device pose on devices without interaction profiles
+        // for hands, such as Varjo and Quest. Should be removed once we have universal
+        // hand interaction profile(s) across vendors.
+        public bool TryGetPolyfillDevicePose(out Pose devicePose)
+        {
+            Handedness handedness = HandNode.ToHandedness();
+            HandJointPoseSource palmPoseSource = new HandJointPoseSource(handedness, TrackedHandJoint.Palm);
```

Comment:
If we only ever care about `Palm` here, should we just use `handsAggregator.TryGetJoint`?

Reply:
We can! Defaulting to using a pose source was more of a code cleanliness thing. 

I'll change it to TryGetJoint to remove any memory concerns.